src/main/java/com/redislabs/redisgraph/impl/api/ContextedRedisGraph.java

```diff
@@ -50,6 +51,9 @@ protected ResultSet sendQuery(String graphId, String preparedQuery) {
             List<Object> rawResponse = (List<Object>) conn.sendCommand(RedisGraphCommand.QUERY, graphId, preparedQuery, Utils.COMPACT_STRING);
             return new ResultSetImpl(rawResponse, this, caches.getGraphCache(graphId));
         }
+        catch (JedisDataException j) {
+            throw new JRedisGraphCompileTimeError(j);
```

Comment:
1. please close the connection `conn.close();`
2. since `JRedisGraphRunTimeError` extends `JedisDataException` the catch clause will catch it. write additional catch for this type, and just throw. Write it before the `JedisDataException` catch, since it inherits from it.